### PR TITLE
Create common action buttons

### PR DESCRIPTION
**Pull request by mob-dev**
- Create common action buttons

┆Issue is synchronized with this [Jira Task](https://highvibe.atlassian.net/browse/HVN1-205) by [Unito](https://www.unito.io/learn-more)

### CODE REVIEW

File: mobile/lib/widgets/buttons/common_action_button.dart [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import 'package:core/common/assets/assets.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+
+class AppBarButtons {
```
Comment (AlexVegner) [inline mobile/lib/widgets/buttons/common_action_button.dart:5]: Please follow task https://github.com/HighVibe/hvn/issues/342

- [ ] Move task to core/lib/widgets/buttons/common_action_button.dart

- [ ] Rename with CommonActionButton

- [ ] We need to have reusable widget. 

AppBarButtons(). back(...) looks ugly. 
Please use main widget constructor + Factory constructors which use defoault costructor. 
But we need a widget which have one animation for all cases

**Review by AlexVegner — CHANGES_REQUESTED** [review]

File: mobile/lib/modules/home/home_page.dart [hunk 2]
```diff
@@ -138,24 +139,39 @@ class _HomePageState extends ModularState<HomePage, HomeController>
                 child: Row(
                   mainAxisAlignment: MainAxisAlignment.spaceBetween,
                   children: <Widget>[
-                    IconButton(
-                      onPressed: _openDrawer,
-                      icon: SvgPicture.asset(Assets.svg.icHamburger),
+                    CommonActionButton(
+                      icon: SvgPicture.asset(Assets.svg.icHamburger) ,
```
Comment (AlexVegner) [inline mobile/lib/modules/home/home_page.dart:143]: Please format the code properly, unnesesary space  `) ,` should be `),`
Use default dart formating onFileSave

File: mobile/lib/modules/home/home_page.dart [hunk 3]
```diff
@@ -138,24 +139,39 @@ class _HomePageState extends ModularState<HomePage, HomeController>
...
+                 /*   AppBarButtons().drawer(
+                      onPressed: _openDrawer,
+                    ),*/
                     Row(
                       children: [
-                        IconButton(
-                          onPressed: () => ChangeThemeModule.toChangeTheme(),
-                          icon: SvgPicture.asset(Assets.svg.icTheme),
+                        CommonActionButton(
+                          icon: SvgPicture.asset(Assets.svg.icTheme) ,
```
Comment (AlexVegner) [inline mobile/lib/modules/home/home_page.dart:152]: Please format the code properly, unnesesary space `) ,` should be `),`
Use default dart formating onFileSave

File: mobile/lib/modules/home/home_page.dart [hunk 4]
```diff
@@ -138,24 +139,39 @@ class _HomePageState extends ModularState<HomePage, HomeController>
...
                   mainAxisAlignment: MainAxisAlignment.spaceBetween,
                   children: <Widget>[
-                    IconButton(
-                      onPressed: _openDrawer,
-                      icon: SvgPicture.asset(Assets.svg.icHamburger),
+                    CommonActionButton(
+                      icon: SvgPicture.asset(Assets.svg.icHamburger) ,
+                      onClickCallback: _openDrawer,
                     ),
+                 /*   AppBarButtons().drawer(
```
Comment (AlexVegner) [inline mobile/lib/modules/home/home_page.dart:146]: Remove commented code

File: mobile/lib/modules/home/home_page.dart [hunk 5]
```diff
@@ -138,24 +139,39 @@ class _HomePageState extends ModularState<HomePage, HomeController>
...
-                          onPressed: () => ChangeThemeModule.toChangeTheme(),
-                          icon: SvgPicture.asset(Assets.svg.icTheme),
+                        CommonActionButton(
+                          icon: SvgPicture.asset(Assets.svg.icTheme) ,
+                          onClickCallback: ChangeThemeModule.toChangeTheme,
                         ),
-                        IconButton(
-                          onPressed: () {},
-                          icon: SvgPicture.asset(Assets.svg.icWallet),
+                  /*      AppBarButtons().themes(
```
Comment (AlexVegner) [inline mobile/lib/modules/home/home_page.dart:155]: Remove commented code

File: mobile/lib/modules/home/home_page.dart [hunk 6]
```diff
@@ -138,24 +139,39 @@ class _HomePageState extends ModularState<HomePage, HomeController>
...
+                          onPressed: ChangeThemeModule.toChangeTheme,
+                        ),*/
+                        CommonActionButton(
+                          icon: SvgPicture.asset(Assets.svg.icWallet) ,
```
Comment (AlexVegner) [inline mobile/lib/modules/home/home_page.dart:159]: Please format the code properly, unnesesary space `) ,` should be `),`
Use default dart formating onFileSave

File: mobile/lib/modules/home/home_page.dart [hunk 7]
```diff
@@ -138,24 +139,39 @@ class _HomePageState extends ModularState<HomePage, HomeController>
...
+                          onClickCallback: (){},
                         ),
-                        IconButton(
-                          onPressed: () {},
-                          icon: SvgPicture.asset(Assets.svg.icNotification),
+                   /*     AppBarButtons().wallet(onPressed: () {
```
Comment (AlexVegner) [inline mobile/lib/modules/home/home_page.dart:162]: Remove commented code

File: mobile/lib/modules/home/home_page.dart [hunk 8]
```diff
@@ -138,24 +139,39 @@ class _HomePageState extends ModularState<HomePage, HomeController>
...
+                          // TODO: Wallet
+                        }
+                        ),*/
+                        CommonActionButton(
+                          icon: SvgPicture.asset(Assets.svg.icNotification) ,
```
Comment (AlexVegner) [inline mobile/lib/modules/home/home_page.dart:167]: Please format the code properly, unnesesary space `) ,` should be `),`
Use default dart formating onFileSave

File: mobile/lib/modules/home/home_page.dart [hunk 9]
```diff
@@ -138,24 +139,39 @@ class _HomePageState extends ModularState<HomePage, HomeController>
...
+                          onClickCallback: (){},
```
Comment (AlexVegner) [inline mobile/lib/modules/home/home_page.dart:168]: Space required `(){}` should be `() {}`

File: mobile/lib/modules/home/home_page.dart [hunk 10]
```diff
@@ -138,24 +139,39 @@ class _HomePageState extends ModularState<HomePage, HomeController>
...
                         ),
+                    /*    AppBarButtons().notification(
```
Comment (AlexVegner) [inline mobile/lib/modules/home/home_page.dart:170]: Remove commented code

File: mobile/lib/modules/meditation/starting_bell.dart [hunk 11]
```diff
@@ -17,12 +18,9 @@ class _StartingBellState extends State<StartingBell> {
...
       appBar: AppBar(
         title: const Text('Starting Bell'),
-        leading: IconButton(
-          icon: const Icon(
-            Icons.close,
-            color: Colors.white,
-          ),
-          onPressed: () => Navigator.pop(context),
+        leading: CommonActionButton(
+          icon:const Icon(Icons.arrow_back) ,
```
Comment (AlexVegner) [inline mobile/lib/modules/meditation/starting_bell.dart:22]: Please format the code properly, unnesesary space `) ,` should be `),`
Use default dart formating onFileSave

File: mobile/lib/modules/meditation/starting_bell.dart [hunk 12]
```diff
@@ -17,12 +18,9 @@ class _StartingBellState extends State<StartingBell> {
...
+          onClickCallback: ()=>Navigator.pop(context),
```
Comment (AlexVegner) [inline mobile/lib/modules/meditation/starting_bell.dart:23]: Spaces required `()=>Navigator.pop(context),` should be `() => Navigator.pop(context),`

File: mobile/lib/modules/mood_tracker/current_mood.dart [hunk 13]
```diff
@@ -31,14 +32,9 @@ class _CurrentMoodState
...
-        leading: IconButton(
-          icon: const Icon(
-            Icons.arrow_back,
-            color: Colors.white,
-          ),
-          onPressed: () {
-            Modular.to.pop();
-          },
+        leading: CommonActionButton(
+          icon:const Icon(Icons.arrow_back) ,
```
Comment (AlexVegner) [inline mobile/lib/modules/mood_tracker/current_mood.dart:36]: Please format the code properly, unnesesary space `) ,` should be `),`
Use default dart formating onFileSave

File: mobile/lib/modules/mood_tracker/current_mood.dart [hunk 14]
```diff
@@ -31,14 +32,9 @@ class _CurrentMoodState
...
+          onClickCallback: ()=> Modular.to.pop(),
```
Comment (AlexVegner) [inline mobile/lib/modules/mood_tracker/current_mood.dart:37]: Space required `()=> Modular.to.pop(),` should be `() => Modular.to.pop(),`

File: mobile/lib/modules/mood_tracker/edit_mood.dart [hunk 15]
```diff
@@ -31,14 +32,9 @@ class _EditMoodScreenState extends State<EditMoodScreen> {
...
-        leading: IconButton(
-          icon: const Icon(
-            Icons.close,
-            color: Colors.white,
-          ),
-          onPressed: () {
-            Modular.to.pop();
-          },
+        leading: CommonActionButton(
+          icon:const Icon(Icons.arrow_back) ,
```
Comment (AlexVegner) [inline mobile/lib/modules/mood_tracker/edit_mood.dart:36]: Please format the code properly, unnesesary space `) ,` should be `),`
Also spece before `const`
Use default dart formating onFileSave

File: mobile/lib/modules/mood_tracker/edit_mood.dart [hunk 16]
```diff
@@ -31,14 +32,9 @@ class _EditMoodScreenState extends State<EditMoodScreen> {
...
+          onClickCallback: ()=> Modular.to.pop(),
```
Comment (AlexVegner) [inline mobile/lib/modules/mood_tracker/edit_mood.dart:37]: Space required `()=> Modular.to.pop(),` should be `() => Modular.to.pop(),`

File: mobile/lib/modules/mood_tracker/mood_tracker_page.dart [hunk 17]
```diff
@@ -20,12 +21,9 @@ class _MoodTrackerPageState
...
     return Scaffold(
       appBar: AppBar(
-          leading: IconButton(
-            icon: const Icon(
-              Icons.close,
-              color: Colors.white,
-            ),
-            onPressed: () => Modular.to.pop(),
+          leading: CommonActionButton(
+            icon:const Icon(Icons.arrow_back) ,
```
Comment (AlexVegner) [inline mobile/lib/modules/mood_tracker/mood_tracker_page.dart:25]: Please format the code properly, unnesesary space `) ,` should be `),`
Also spece before `const`
Use default dart formating onFileSave

File: mobile/lib/modules/onboarding/onboarding_page.dart [hunk 18]
```diff
@@ -18,9 +19,9 @@ class _OnboardingUserOptionsState extends ModularState<OnboardingUserOptions,
     return Scaffold(
       backgroundColor: AppColors.primaryColor,
       appBar: AppBar(
-        leading: IconButton(
-          icon: const Icon(Icons.arrow_back, color: Colors.white),
-          onPressed: () => Modular.to.pop(),
+        leading: CommonActionButton(
+          icon:const Icon(Icons.arrow_back),
```
Comment (AlexVegner) [inline mobile/lib/modules/onboarding/onboarding_page.dart:23]: Add space before `const`

File: mobile/lib/modules/playlist/add_to_playlist/add_to_playlist_page.dart [hunk 19]
```diff
@@ -53,10 +54,17 @@ class _AddToPlaylistPageState
...
-          onPressed: _close,
+        leading:
+        CommonActionButton(
+          icon: const Icon(
+            Icons.close,
+            color: Colors.white,
+          ),
+          onClickCallback: _close,
         ),
+    /*    AppBarButtons().close(
```
Comment (AlexVegner) [inline mobile/lib/modules/playlist/add_to_playlist/add_to_playlist_page.dart:65]: Remove commented code

File: mobile/lib/modules/playlist/create_new_playlist/create_new_playlist_page.dart [hunk 20]
```diff
@@ -40,9 +41,9 @@ class _CreateNewPlaylistPage
             fontWeight: FontWeight.bold,
           ),
         ),
-        leading: IconButton(
-          icon: const Icon(Icons.arrow_back, color: Colors.white),
-          onPressed: () => _close(),
+        leading:CommonActionButton(
+          icon:const Icon(Icons.arrow_back) ,
```
Comment (AlexVegner) [inline mobile/lib/modules/playlist/create_new_playlist/create_new_playlist_page.dart:45]: Please format the code properly, unnesesary space `) ,` should be `),`
Also spece before `const`
Use default dart formating onFileSave

File: mobile/lib/modules/profile/current_user/edit_profile/edit_profile_page.dart [hunk 21]
```diff
@@ -30,9 +31,10 @@ class _EditProfilePageState
     return Scaffold(
       appBar: AppBar(
         backgroundColor: Colors.transparent,
-        leading: IconButton(
-            icon: const Icon(Icons.arrow_back, color: Colors.white),
-            onPressed: () => Modular.to.pop()),
+        leading: CommonActionButton(
+          icon:const Icon(Icons.arrow_back) ,
```
Comment (AlexVegner) [inline mobile/lib/modules/profile/current_user/edit_profile/edit_profile_page.dart:35]: Please format the code properly, unnesesary space `) ,` should be `),`
Also spece before `const`
Use default dart formating onFileSave

File: mobile/lib/modules/profile/current_user/upload_audio/upload_audio_page.dart [hunk 22]
```diff
@@ -38,9 +39,9 @@ class _UploadAudioPageState
           'Upload Audio',
           style: TextStyle(fontWeight: FontWeight.bold),
         ),
-        leading: IconButton(
-          icon: const Icon(Icons.cancel, color: Colors.white),
-          onPressed: () => Modular.to.pop(),
+        leading: CommonActionButton(
+          icon:const Icon(Icons.cancel, color: Colors.white) ,
```
Comment (AlexVegner) [inline mobile/lib/modules/profile/current_user/upload_audio/upload_audio_page.dart:43]: Please format the code properly, unnesesary space `) ,` should be `),`
Also spece before `const`
Use default dart formating onFileSave

File: mobile/lib/modules/profile/users_list/users_list_page.dart [hunk 23]
```diff
@@ -32,8 +33,9 @@ class _UsersListPageState
       appBar: AppBar(
         title: Text(widget.title),
         actions: [
-          BackButton(
-            onPressed: Modular.to.maybePop,
+          CommonActionButton(
+            icon:const Icon(Icons.arrow_back) ,
```
Comment (AlexVegner) [inline mobile/lib/modules/profile/users_list/users_list_page.dart:37]: Please format the code properly, unnesesary space `) ,` should be `),`
Also spece before `const`
Use default dart formating onFileSave

File: mobile/lib/modules/wallet/pages/wallet/wallet_restore.dart [hunk 24]
```diff
@@ -27,12 +28,9 @@ class _WalletRestoreState
...
     return Scaffold(
       appBar: AppBar(
-        leading: IconButton(
-          icon: const Icon(
-            Icons.arrow_back,
-            color: Colors.white,
-          ),
-          onPressed: () => Modular.to.pop(),
+        leading: CommonActionButton(
+          icon:const Icon(Icons.arrow_back) ,
```
Comment (AlexVegner) [inline mobile/lib/modules/wallet/pages/wallet/wallet_restore.dart:32]: Please format the code properly, unnesesary space `) ,` should be `),`
Also spece before `const`
Use default dart formating onFileSave

File: mobile/lib/modules/wallet/pages/wallet/wallet_seed_phrase.dart [hunk 25]
```diff
@@ -26,9 +27,10 @@ class _WalletSeedPhraseState
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(
-        leading: IconButton(
-            icon: const Icon(Icons.arrow_back, color: Colors.white),
-            onPressed: () => Navigator.of(context).pop()),
+        leading:CommonActionButton(
+          icon:const Icon(Icons.arrow_back) ,
```
Comment (AlexVegner) [inline mobile/lib/modules/wallet/pages/wallet/wallet_seed_phrase.dart:31]: Please format the code properly, unnesesary space `) ,` should be `),`
Also spece before `const`
Use default dart formating onFileSave

File: mobile/lib/modules/wallet/pages/wallet/wallet_seed_phrase.dart [hunk 26]
```diff
@@ -26,9 +27,10 @@ class _WalletSeedPhraseState
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(
-        leading: IconButton(
-            icon: const Icon(Icons.arrow_back, color: Colors.white),
-            onPressed: () => Navigator.of(context).pop()),
+        leading:CommonActionButton(
```
Comment (AlexVegner) [inline mobile/lib/modules/wallet/pages/wallet/wallet_seed_phrase.dart:30]: Add space

File: mobile/lib/modules/wallet/pages/wallet/wallet_seed_phrase.dart [hunk 27]
```diff
@@ -26,9 +27,10 @@ class _WalletSeedPhraseState
...
+          icon:const Icon(Icons.arrow_back) ,
+          onClickCallback: ()=> Modular.to.pop(),
```
Comment (AlexVegner) [inline mobile/lib/modules/wallet/pages/wallet/wallet_seed_phrase.dart:32]: Add space

File: mobile/lib/modules/wallet/pages/wallet/wallet_terms_and_conditions.dart [hunk 28]
```diff
@@ -19,9 +20,9 @@ class _WalletTermsAndConditionsState extends State<WalletTermsAndConditions> {
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(
-        leading: IconButton(
-          icon: const Icon(Icons.arrow_back, color: Colors.white),
-          onPressed: () => Modular.to.pop(),
+        leading: CommonActionButton(
+          icon:const Icon(Icons.arrow_back) ,
```
Comment (AlexVegner) [inline mobile/lib/modules/wallet/pages/wallet/wallet_terms_and_conditions.dart:24]: Please format the code properly, unnesesary space `) ,` should be `),`
Also spece before `const`
Use default dart formating onFileSave

File: mobile/lib/modules/wallet/pages/wallet/wallet_terms_and_conditions.dart [hunk 29]
```diff
@@ -19,9 +20,9 @@ class _WalletTermsAndConditionsState extends State<WalletTermsAndConditions> {
...
+          onClickCallback: ()=> Modular.to.pop(),
```
Comment (AlexVegner) [inline mobile/lib/modules/wallet/pages/wallet/wallet_terms_and_conditions.dart:25]: Add space

File: mobile/lib/modules/wallet/pages/wallet/wallet_verify_seed_phrase.dart [hunk 30]
```diff
@@ -32,12 +33,9 @@ class _WalletVerifySeedPhraseState
...
     return Scaffold(
       appBar: AppBar(
-        leading: IconButton(
-          icon: const Icon(
-            Icons.arrow_back,
-            color: Colors.white,
-          ),
-          onPressed: () => Modular.to.pop(),
+        leading: CommonActionButton(
+          icon:const Icon(Icons.arrow_back) ,
```
Comment (AlexVegner) [inline mobile/lib/modules/wallet/pages/wallet/wallet_verify_seed_phrase.dart:37]: Please format the code properly, unnesesary space `) ,` should be `),`
Also spece before `const`
Use default dart formating onFileSave

File: mobile/lib/modules/wallet/pages/wallet/wallet_verify_seed_phrase.dart [hunk 31]
```diff
@@ -32,12 +33,9 @@ class _WalletVerifySeedPhraseState
...
+          onClickCallback: ()=> Modular.to.pop(),
```
Comment (AlexVegner) [inline mobile/lib/modules/wallet/pages/wallet/wallet_verify_seed_phrase.dart:38]: Add spcae

File: mobile/lib/modules/wallet/processing_transaction_page.dart [hunk 32]
```diff
@@ -9,7 +10,10 @@ class ProcessingTransactionPage extends StatelessWidget {
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(
-        leading: const Icon(Icons.send),
+        leading:  CommonActionButton(
+          icon: const Icon(Icons.send),
+          onClickCallback: (){},
```
Comment (AlexVegner) [inline mobile/lib/modules/wallet/processing_transaction_page.dart:15]: Add space

File: mobile/lib/modules/wallet/qrcode_reader_page.dart [hunk 33]
```diff
@@ -158,11 +159,9 @@ class _QRCodeReaderPageState extends State<QRCodeReaderPage> {
         centerTitle: true,
         backgroundColor: Colors.transparent,
         elevation: 0.0,
-        leading: IconButton(
-          icon: ImageIcon(
-            AssetImage(Assets.images.back),
-          ),
-          onPressed: () => Modular.to.pop(),
+        leading: CommonActionButton(
+          icon:const Icon(Icons.arrow_back) ,
```
Comment (AlexVegner) [inline mobile/lib/modules/wallet/qrcode_reader_page.dart:163]: Please format the code properly, unnesesary space `) ,` should be `),`
Also spece before `const`
Use default dart formating onFileSave

File: mobile/lib/modules/wallet/qrcode_reader_page.dart [hunk 34]
```diff
@@ -158,11 +159,9 @@ class _QRCodeReaderPageState extends State<QRCodeReaderPage> {
...
+          onClickCallback: ()=> Modular.to.pop(),
```
Comment (AlexVegner) [inline mobile/lib/modules/wallet/qrcode_reader_page.dart:164]: Add space

File: mobile/lib/modules/wallet/wallet_import_page.dart [hunk 35]
```diff
@@ -37,11 +38,9 @@ class _WalletImportPage
         centerTitle: true,
         backgroundColor: Colors.transparent,
         elevation: 0.0,
-        leading: IconButton(
-          icon: ImageIcon(
-            AssetImage(Assets.images.back),
-          ),
-          onPressed: () => Modular.to.pop(),
+        leading:CommonActionButton(
```
Comment (AlexVegner) [inline mobile/lib/modules/wallet/wallet_import_page.dart:41]: Add space

File: mobile/lib/modules/wallet/wallet_import_page.dart [hunk 36]
```diff
@@ -37,11 +38,9 @@ class _WalletImportPage
...
+          icon:const Icon(Icons.arrow_back) ,
```
Comment (AlexVegner) [inline mobile/lib/modules/wallet/wallet_import_page.dart:42]: Please format the code properly, unnesesary space `) ,` should be `),`
Also spece before `const`
Use default dart formating onFileSave

File: mobile/lib/modules/wallet/wallet_import_page.dart [hunk 37]
```diff
@@ -37,11 +38,9 @@ class _WalletImportPage
...
+          onClickCallback: ()=> Modular.to.pop(),
```
Comment (AlexVegner) [inline mobile/lib/modules/wallet/wallet_import_page.dart:43]: Add space

File: mobile/lib/modules/wallet/wallet_transfer_page.dart [hunk 38]
```diff
@@ -42,12 +42,13 @@ class _WalletTransferPageState
         backgroundColor: Colors.transparent,
         elevation: 0.0,
         actions: <Widget>[
-          IconButton(
-            icon: ImageIcon(
+          CommonActionButton(
+            icon:ImageIcon(
```
Comment (AlexVegner) [inline mobile/lib/modules/wallet/wallet_transfer_page.dart:46]: Add space

File: mobile/lib/modules/wallet/pages/wallet/wallet_restore.dart [hunk 39]
```diff
@@ -27,12 +28,9 @@ class _WalletRestoreState
...
+          onClickCallback: ()=> Modular.to.pop(),
```
Comment (AlexVegner) [inline mobile/lib/modules/wallet/pages/wallet/wallet_restore.dart:33]: Add space

File: mobile/lib/modules/profile/users_list/users_list_page.dart [hunk 40]
```diff
@@ -32,8 +33,9 @@ class _UsersListPageState
...
+            onClickCallback: ()=> Modular.to.pop(),
```
Comment (AlexVegner) [inline mobile/lib/modules/profile/users_list/users_list_page.dart:38]: Add space

File: mobile/lib/modules/profile/current_user/upload_audio/upload_audio_page.dart [hunk 41]
```diff
@@ -38,9 +39,9 @@ class _UploadAudioPageState
...
+          onClickCallback: ()=> Modular.to.pop(),
```
Comment (AlexVegner) [inline mobile/lib/modules/profile/current_user/upload_audio/upload_audio_page.dart:44]: Add space

File: mobile/lib/modules/profile/current_user/edit_profile/edit_profile_page.dart [hunk 42]
```diff
@@ -30,9 +31,10 @@ class _EditProfilePageState
...
+          onClickCallback: ()=> Modular.to.pop(),
```
Comment (AlexVegner) [inline mobile/lib/modules/profile/current_user/edit_profile/edit_profile_page.dart:36]: Add space

File: mobile/lib/modules/profile/current_user/edit_audio/edit_audio_page.dart [hunk 43]
```diff
@@ -29,9 +30,9 @@ class _EditAudioPageState
       key: _scaffoldKey,
       appBar: AppBar(
         backgroundColor: Colors.transparent,
-        leading: IconButton(
-          icon: const Icon(Icons.arrow_back, color: Colors.white),
-          onPressed: () => Modular.to.pop(),
+        leading: CommonActionButton(
+          icon:const Icon(Icons.arrow_back) ,
```
Comment (AlexVegner) [inline mobile/lib/modules/profile/current_user/edit_audio/edit_audio_page.dart:34]: Please format the code properly, unnesesary space `) ,` should be `),`
Also spece before `const`
Use default dart formating onFileSave

File: mobile/lib/modules/profile/current_user/edit_audio/edit_audio_page.dart [hunk 44]
```diff
@@ -29,9 +30,9 @@ class _EditAudioPageState
...
+          onClickCallback: ()=> Modular.to.pop(),
```
Comment (AlexVegner) [inline mobile/lib/modules/profile/current_user/edit_audio/edit_audio_page.dart:35]: Add space

File: mobile/lib/modules/playlist/edit_playlist/edit_playlist_page.dart [hunk 45]
```diff
@@ -37,9 +38,10 @@ class _EditPlayListPageState
             fontWeight: FontWeight.bold,
           ),
         ),
-        leading: IconButton(
-            icon: const Icon(Icons.arrow_back, color: Colors.white),
-            onPressed: () => Modular.to.pop()),
+        leading:CommonActionButton(
+          icon:const Icon(Icons.arrow_back) ,
```
Comment (AlexVegner) [inline mobile/lib/modules/playlist/edit_playlist/edit_playlist_page.dart:42]: Please format the code properly, unnesesary space `) ,` should be `),`
Also spece before `const`
Use default dart formating onFileSave

File: mobile/lib/modules/playlist/edit_playlist/edit_playlist_page.dart [hunk 46]
```diff
@@ -37,9 +38,10 @@ class _EditPlayListPageState
             fontWeight: FontWeight.bold,
           ),
         ),
-        leading: IconButton(
-            icon: const Icon(Icons.arrow_back, color: Colors.white),
-            onPressed: () => Modular.to.pop()),
+        leading:CommonActionButton(
```
Comment (AlexVegner) [inline mobile/lib/modules/playlist/edit_playlist/edit_playlist_page.dart:41]: Add space

File: mobile/lib/modules/playlist/edit_playlist/edit_playlist_page.dart [hunk 47]
```diff
@@ -37,9 +38,10 @@ class _EditPlayListPageState
...
+          icon:const Icon(Icons.arrow_back) ,
+          onClickCallback: ()=> Modular.to.pop(),
```
Comment (AlexVegner) [inline mobile/lib/modules/playlist/edit_playlist/edit_playlist_page.dart:43]: Add space

File: mobile/lib/modules/playlist/create_new_playlist/create_new_playlist_page.dart [hunk 48]
```diff
@@ -40,9 +41,9 @@ class _CreateNewPlaylistPage
             fontWeight: FontWeight.bold,
           ),
         ),
-        leading: IconButton(
-          icon: const Icon(Icons.arrow_back, color: Colors.white),
-          onPressed: () => _close(),
+        leading:CommonActionButton(
```
Comment (AlexVegner) [inline mobile/lib/modules/playlist/create_new_playlist/create_new_playlist_page.dart:44]: Add space

File: mobile/lib/modules/onboarding/onboarding_page.dart [hunk 49]
```diff
@@ -18,9 +19,9 @@ class _OnboardingUserOptionsState extends ModularState<OnboardingUserOptions,
...
+          onClickCallback: ()=> Modular.to.pop(),
```
Comment (AlexVegner) [inline mobile/lib/modules/onboarding/onboarding_page.dart:24]: Add space

File: mobile/lib/modules/mood_tracker/mood_tracker_page.dart [hunk 50]
```diff
@@ -20,12 +21,9 @@ class _MoodTrackerPageState
...
+            onClickCallback: ()=> Modular.to.pop(),
```
Comment (AlexVegner) [inline mobile/lib/modules/mood_tracker/mood_tracker_page.dart:26]: Add space

File: mobile/lib/modules/auth/register/register_with_sms/create_password/create_password_page.dart [hunk 51]
```diff
@@ -27,10 +28,14 @@ class _CreatePasswordPageState
...
       backgroundColor: AppColors.primaryColor,
       appBar: AppBar(
-        leading: IconButton(
-          icon: const Icon(Icons.arrow_back),
+        leading: CommonActionButton(
+          icon:const Icon(Icons.arrow_back) ,
+          onClickCallback: ()=> Modular.to.pop(),
+        )
+
+      /*  AppBarButtons().back(
```
Comment (AlexVegner) [inline mobile/lib/modules/auth/register/register_with_sms/create_password/create_password_page.dart:36]: Remove commented code

File: core/lib/widgets/buttons/common_action_button.dart [hunk 52]
```diff
@@ -0,0 +1,75 @@
+import 'package:flutter/material.dart';
+
+class CommonActionButton extends StatefulWidget {
+
+  CommonActionButton({this.icon, this.color, this.onClickCallback}) : super();
+
+  final Widget icon;
+  final Color color;
+  final OnClickCallback onClickCallback;
```
Comment (AlexVegner) [inline core/lib/widgets/buttons/common_action_button.dart:9]: Use VoidCallback from Flutter SDK, use simple name like Flitted do `onPressed`, `onTap`, `onClick` 
It is obvious that this is a callback.

**Review by AlexVegner — CHANGES_REQUESTED** [review]

File: mobile/.gitignore [hunk 53]
```diff
@@ -21,6 +21,12 @@ report.xml
 *.ipr
 *.iws
 .idea/
+local.properties
+gradlew.bat
```
Comment (AlexVegner) [inline mobile/.gitignore:25]: We need mobile/gradle/wrapper folder. 
So please fix it.

File: mobile/.gitignore [hunk 54]
```diff
@@ -21,6 +21,12 @@ report.xml
...
+gradlew
+build.gradle
```
Comment (AlexVegner) [inline mobile/.gitignore:27]: We need build.gradle file under  mobile/android and mobile/android/app, so please fix it.

File: mobile/.gitignore [hunk 55]
```diff
@@ -21,6 +21,12 @@ report.xml
 *.ipr
 *.iws
 .idea/
+local.properties
```
Comment (AlexVegner) [inline mobile/.gitignore:24]: Check mobile/android/.gitignore it already have those rules.

File: .gitignore [hunk 56]
```diff
@@ -45,6 +44,14 @@ report.xml
 *.ipr
 *.iws
 .idea/
+.gradle/
```
Comment (AlexVegner) [inline .gitignore:47]: Please make sure that those rules will relate only root folder.
sample
```
/.gradle
/.gradle
/gradlew
/gradlew.bat
/local.properties
```

**Review by AlexVegner — CHANGES_REQUESTED** [review]

File: mobile/.gitignore (hunk 53 above)
Comment (mob-dev) [reply mobile/.gitignore:25]: You mean, mobile/android/gradle/wrapper folder. Right?

**Review by AlexVegner — APPROVED** [review]

**Pull request merged**